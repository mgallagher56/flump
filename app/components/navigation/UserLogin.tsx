import type { FC } from 'react';

import { useLoaderData } from '@remix-run/react';
import { useTranslation } from 'react-i18next';
import type { loader } from '~/root';

import FLPButtonGroup from '../core/buttons/FLPButtonGroup';
import FLPLinkButton from '../core/buttons/FLPLinkButton';
import SignOut from '../users/SignOut';

const UserLogin: FC = () => {
  const { t } = useTranslation();
  const { user } = useLoaderData<typeof loader>();

  return (
    <FLPButtonGroup>
      {user ? (
        <SignOut />
      ) : (
        <>
          <FLPLinkButton to="/signup" variant="solid">
            {t('signUp')}
          </FLPLinkButton>
          <FLPLinkButton to="/login" variant="outline">
            {t('logIn')}
          </FLPLinkButton>
        </>
      )}
    </FLPButtonGroup>
  );
};

export default UserLogin;
