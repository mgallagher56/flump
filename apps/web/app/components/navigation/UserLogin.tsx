import type { FC } from 'react';

import { useTranslation } from 'react-i18next';
import { useLoaderData } from 'react-router';
import FLPButtonGroup from '~/components/core/buttons/FLPButtonGroup';
import FLPLinkButton from '~/components/core/buttons/FLPLinkButton';
import SignOut from '~/components/users/SignOut';
import type { loader } from '~/root';

const UserLogin: FC = () => {
  const { t } = useTranslation();

  const { user } = useLoaderData<typeof loader>();

  return (
    <FLPButtonGroup>
      {user ? (
        <SignOut />
      ) : (
        <>
          <FLPLinkButton to="/signup" variant="solid">
            {t('signUp')}
          </FLPLinkButton>
          <FLPLinkButton to="/login" variant="outline">
            {t('logIn')}
          </FLPLinkButton>
        </>
      )}
    </FLPButtonGroup>
  );
};

export default UserLogin;
