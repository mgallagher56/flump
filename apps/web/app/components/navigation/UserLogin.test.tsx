import { render } from '@testing-library/react';
import { vi } from 'vitest';

import mockUser from '__mocks__/user';

import UserLogin from './UserLogin';

const mocks = vi.hoisted(() => ({
  mockUseLoaderData: vi.fn()
}));

vi.mock('react-router', async () => {
  const actual: Record<string, unknown> = await vi.importActual('react-router');
  return {
    ...actual,
    useLoaderData: mocks.mockUseLoaderData
  };
});

vi.mock('app/components/users/SignOut', () => ({
  default: () => <div>logOut</div>
}));

vi.mock('app/components/core/buttons/FLPLinkButton', () => ({
  default: (props) => <div>{props.children}</div>
}));

describe('<UserLogin />', () => {
  test('renders as expected when user not logged in', () => {
    mocks.mockUseLoaderData.mockReturnValue({ user: undefined });
    const { baseElement } = render(<UserLogin />);
    expect(baseElement).toMatchSnapshot();
  });
  test('renders signed out when logged in', () => {
    mocks.mockUseLoaderData.mockReturnValue({ user: mockUser });
    const { getByText } = render(<UserLogin />);
    expect(getByText('logOut')).toBeDefined();
  });
});
