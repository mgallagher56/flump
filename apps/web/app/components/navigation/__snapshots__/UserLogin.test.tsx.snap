// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<UserLogin /> > renders as expected when user not logged in 1`] = `
<body>
  <div>
    <script
      nonce=""
    >
      ((e,r,s,u,d,m,l,h)=&gt;
      {let c=document.documentElement,v=["light","dark"];function p(i){(Array.isArray(e)?e:[e]).forEach(y=&gt;
      {let k=y==="class",S=k&&m?d.map(f=&gt;
      m[f]||f):d;k?(c.classList.remove(...S),c.classList.add(i)):c.setAttribute(y,i)}),R(i)}function R(i){h&&v.includes(i)&&(c.style.colorScheme=i)}function a(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let i=localStorage.getItem(r)||s,y=l&&i==="system"?a():i;p(y)}catch(i){}})("class","theme","system",null,["light","dark"],null,true,true)
    </script>
    <div
      class="css-6l0f26"
    >
      <div />
      <div />
    </div>
  </div>
</body>
`;
